pkg/binary/envoy/debug/lsof.go

```diff
@@ -55,33 +58,46 @@ func EnableOpenFilesDataCollection(r *envoy.Runtime) {
 }
 
 // retrieveOpenFilesData writes statistics of open files associated with envoy instance(s) to a json file
-// if succeeded, return nil, else return an error instance
+// Errors from platform-specific libraries log to debug instead of raising an error or logging in an error category.
+// This avoids filling logs for unresolvable reasons.
 func retrieveOpenFilesData(r binary.Runner) error {
-	// get pid of envoy instance
-	pid, err := r.GetPid()
+	f, err := os.Create(filepath.Join(r.DebugStore(), "lsof/lsof.json"))
 	if err != nil {
-		return fmt.Errorf("error in getting pid of envoy instance: %v", err)
+		return fmt.Errorf("error in creating a file to write open file statistics to: %w", err)
 	}
-	envoyProcess, err := process.NewProcess(int32(pid))
+	defer f.Close() //nolint
+
+	// get pid of envoy instance
+	p, err := r.GetPid()
 	if err != nil {
-		return fmt.Errorf("error in creating an envoy instance: %v", err)
+		return fmt.Errorf("error in getting pid of envoy instance: %w", err)
 	}
+	pid := int32(p)
 
-	f, err := os.Create(filepath.Join(r.DebugStore(), "lsof/lsof.json"))
-	if err != nil {
-		return fmt.Errorf("error in creating a file to write open file statistics to: %v", err)
+	ctx, cancel := context.WithTimeout(context.Background(), processTimeout)
```

Comment:
we have to use the XXXWithContext functionality here, else things can hang forever